statstool

```diff
@@ -106,891 +102,118 @@ echo "we are only going to read data from your repo and GitHub"
-        GITHUB_TOKEN=$GITHUB_TOKEN HEADER_ACCEPT="application/vnd.github.groot-preview+json" PREFIX_URL=true SILENT=true ${DIR}/fetch-comments.sh | pv -l -s200 \
-        > ${DATADIR}/commit_pulls
-    github_commit_pull_time=$(( $(date +%s) - ${starttime}))
-    echo "done pulling commit pull request info in ${github_commit_pull_time}s."
-    echo
-    printf "github_commit_pull_time\t%f\n" ${github_commit_pull_time} >> ${DATADIR}/times.tsv
-
-    if ! (cat ${DATADIR}/pulls.gz | zcat | head -n1 || true) | grep ',    "number": [0-9]*' > /dev/null; then
-        echo "there are no PRs in this repository"
-        echo
-
-        #just leave the sample files empty
-        touch ${DATADIR}/pr_sample
-        touch ${DATADIR}/pr_sample_pulls
-        touch ${DATADIR}/pr_sample_commits
-    else
-        #get a sample of pulls to prep some additional data
-        echo "preparing additional per-PR data for a ${PR_SAMPLESIZE} PR sample of PRs..."
+fi
 
-        starttime=$(date +%s)
-        echo "drawing the sample..."
-        pv ${DATADIR}/pulls.gz | zcat | grep -o ',    "number": [0-9]*' | sed -e 's/.* \([0-9]*\)$/\1/' | sort -R | tail -n${PR_SAMPLESIZE} | LC_ALL=C sort > ${DATADIR}/pr_sample
+REPO=$(echo "$ORIGIN_URL" | sed 's/\.git$//' | sed -E '/^(git@|https:\/\/([^:@/]*(:[^:@]*)?@)?)github.com[:\/]/!{q1}; {s/.*github.com[:\/]([^\/]*\/[^\]*)$/\1/}')
 
-        echo "fetching full pull objects for PR sample (this might take a while)..."
-        cat ${DATADIR}/pr_sample | gawk '{printf("https://api.github.com/repos/'${REPO}'/pulls/%d?per_page=100\n", $1)}' | GITHUB_TOKEN=$GITHUB_TOKEN ALLCOMMENTS="" SILENT=true ${DIR}/fetch-comments.sh | pv -l -s$(cat ${DATADIR}/pr_sample | wc -l) > ${DATADIR}/pr_sample_pulls
+echo "using remote url ${ORIGIN_URL}"
+echo "this corresponds to the github repository at https://github.com/${REPO}"
+echo "this is the repository whose pull requests and issues we'll analyze"
+echo
 
-        echo "fetch commit list for PR sample (this might take a while)..."
-    cat ${DATADIR}/pr_sample | gawk '{printf("https://api.github.com/repos/'${REPO}'/pulls/%d/commits?per_page=100\n", $1)}' | GITHUB_TOKEN=$GITHUB_TOKEN ALLCOMMENTS="" SILENT=true PREFIX_URL=true ${DIR}/fetch-comments.sh | pv -l -s$(cat ${DATADIR}/pr_sample | wc -l) > ${DATADIR}/pr_sample_commits
+if [ ! -z "${DATADIR}" ]; then
+    echo "using given path to hold temp data"
+    mkdir "${DATADIR}"
+fi
+echo "we'll write data to ${DATADIR}"
+echo
 
-        github_pull_sample_time=$(( $(date +%s) - ${starttime}))
-        echo "done collecting sample PR data in ${github_pull_sample_time}s."
-        echo
-        printf "github_pull_sample_time\t%f\n" ${github_pull_sample_time} >> ${DATADIR}/times.tsv
+#truncate timing data
+> ${DATADIR}/times.tsv
+
+#record when we start pulling data
+#this can be used as a good timestamp for when data collection happened
+printf "data_pull_start_time\t%d\n" $data_pull_start_time >> ${DATADIR}/times.tsv
```

Comment:
It's now getting passed as an environment variable to both scripts.
I can change the formatting to be SCREAMING_SNAKE_CASE, but I just wanted to avoid changing too much what I didn't need to.

Reply:
that's sort of a weird interface choice since the outer script doesn't get any feedback about it and it won't change the behavior in any way the caller can see. But if you like it this way, I won't complain any more.

Reply:
I agree that it's weird but I was trying to preserve the behavior as much as possible.
I could alternatively just move it out of the statstool and move it to fetch_github_data.sh.
I think the chunk of code it precedes doesn't really take too long (unless we're rate limited) so it shouldn't really skew the times much.

Reply:
I see, you're trying to still measure the end-to-end time, now across both scripts. I think you can probably just nuke this stat altogether. But, not critical either way.

Reply:
Made a note for myself to clean this up after this first push.